.github/workflows/CI-unittests.yml

```diff
@@ -101,6 +101,12 @@ jobs:
         run: |
           pytest --durations=10 --color=yes -n auto --dist loadscope
 
+      - name: Random order
```

Comment:
One question: is this going to be applied only to the main or to each branch?

Reply:
Yes, only to main. In 99% of all cases, tests don't depend on their sequence. For the rare cases for which they do, it is better good enough to find this out after merge-to-main.

To add: this is only used for the scheduled runs which runs automatically every night on the main branch.